src/monoclr/pynetinit.c

```diff
@@ -19,6 +19,11 @@ PyNet_Args *PyNet_Init(int ext)
     pn_args->shutdown = NULL;
     pn_args->module = NULL;
 
+#ifdef __linux__
+    // Force preload libmono-2.0 as global
+    dlopen("libmono-2.0.so", RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
Why is this necessary?

Reply:
That is what fixes #1034. The issue seems to be that `libmono` doesn't seem to be globally loaded when `libmononative` is loaded (by `libmono` itself). Doing that here explicitly and leaking the handle ensures that the required symbols are available. Check the recent discussion on the issue.

Reply:
Can you link the relevant post in code, or quote it there?

Reply:
https://github.com/pythonnet/pythonnet/issues/1034#issuecomment-626369097 and https://github.com/pythonnet/pythonnet/issues/1034#issuecomment-626373989.

Reply:
@filmor sorry, I meant put the links/quotes as comments into code.

Reply:
I see. I'll extend the comment, I don't think there is much value in the individual links (will be essentially linked via the commit message).

Reply:
Thx! It is way better to read through code without "wtf is this here for" when you are debugging :-D